app/models/mapping.rb

```diff
@@ -1,24 +1,32 @@
 require 'etc'
 require 'fileutils'
 require 'ood_support'
+require 'ostruct'
 require 'pathname'
 require 'yaml/store'
 
 
 class Mapping < ActiveRecord::Base
   attr_accessor :save_message
-  validates :user, :app, :dataset, presence: true
+  validates :user, :app, :dataset, presence: { message: "A %{attribute} must be selected." }
   validate :dataset_path_must_exist
+  validate :app_path_may_not_be_blank
   validates_uniqueness_of :user, scope: [:user, :app], message: "Unable to create a second mapping between user and app."
 
+  def initialize(params)
+    super(params)
+
+    @save_message = ''
+  end
+
   # Type dataset as a Pathname
   def dataset
-    Pathname.new(super)
+    super ? Pathname.new(super): Pathname.new('')
```

Comment:
1. You can accomplish the same thing doing this:

   ```ruby
   Pathname.new(super.to_s)
   ```

   since `nil.to_s == ''`

2. `Pathname.new('')` is not a great "null object".  This is a difficult thing with Ruby that I don't have a good answer to right now.  Notice:

```ruby
irb(main):012:0> Pathname.new('').to_s
=> ""
irb(main):013:0> Pathname.new('').exist?
=> false
irb(main):014:0> Pathname.new('').file?
=> false
irb(main):015:0> Pathname.new('').directory?
=> false
irb(main):016:0> Pathname.new('').expand_path
=> #<Pathname:/Users/efranz>
irb(main):016:0> Pathname.new('').expand_path.exist?
=> true
irb(main):025:0> Pathname.new('').expand_path.directory?
=> true
```

And I might get clever and suggest Pathname.new('/dev/null') but then this has its own problems:

```
irb(main):022:0> Pathname.new('/dev/null').expand_path
=> #<Pathname:/dev/null> # good!
irb(main):018:0> Pathname.new('/dev/null').directory?
=> false  # good!
irb(main):019:0> Pathname.new('/dev/null').file?
=> false  # good!
irb(main):020:0> Pathname.new('/dev/null').exist?
=> true  # bad!
```

Of course, nil is normally not desired, because then you have to add guard clauses everywhere and nil is generally a bad idea. But in this case I wonder if we should use it for the "null object", unless there is a sensible null object for Pathname that can be produced (maybe we could make our own? - like Pathname.new('') and then override expand_path and other methods that return values we don't like).

Reply:
I was looking for a default constructor or invalid test for `Pathname`. Using `/dev/null` is a nice idea, except for its perfectly correct `exist?` behavior.